### DISCUSSION TITLE
Labeler action - on submodule SHA update

### DISCUSSION THREAD

**Discussion by crissuciu** [community/community#32945] [Actions]
I need the labeler to flag out changes to our submodule SHA.

No regex like this was picked up

`"Submodule": - any: ['submodule-folder/**/*', '*submodule-folder*', './submodule-folder/.*)']`

Any idea how I can solve this?

**Comment by airtower-luna** [comment]
I'm not familiar with `actions/labeler`, but for Git in general a change to a submodule commit is like a change to a file named like the the submodule directory. :thinking: 

Have you tried `'submodule-folder'` as a pattern?

**Reply by crissuciu** [reply]
@airtower-luna I'm just starting off too with this, do you have any documentation links related to this 'submodule-folder'?

**Reply by airtower-luna** [reply]
When you add a submodule to your repository it creates a folder (or directory) to hold the submodule. But what gets added to the index is a file that describes at which commit the submodule is, and an entry in `.gitmodules` (see [Git Tools - Submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules)).

So if there's a directory `my-submodule/` in your working tree that holds a submodule, the index will have a file `my-submodule` instead that lists the commit. That file will change if you change the commit.

So what I'm saying is you should write the pattern for `actions/labeler` like you're trying to catch changes to a *file* named like the submodule.

**Comment by crissuciu** [comment]
That's exactly what I was trying to achieve by this 

"Submodule": - any: ['submodule-folder/**/*', '*submodule-folder*', './submodule-folder/.*', 'submodule-folder']

but it doesn't match with the folder name.

**Reply by airtower-luna** [reply]
That's different from your original post, I see `'submodule-folder'` is in there twice now. I'm afraid that looks like labeler doesn't work like regular Git then. :confused: 

Maybe open an issue over there then, or check the code if you have time.

**Reply by crissuciu** [reply]
Yep, I've just tried the 4th expression without any luck.
I've already submitted an issue for this https://github.com/github/issue-labeler/issues/54

**Reply by airtower-luna** [reply]
I don't think that's the right repository, that's about labeling *issues*, not PRs based on the changes they introduced. Which action do you actually use? If it's `github/issue-labeler` that might actually be your problem.

Also the issue description is empty, without details you're unlikely to get what you want even in the right repository. :sweat_smile:

**Reply by crissuciu** [reply]
Haha, you're right! Thanks for catching this! I'm using actions/labeler@v3
